### PR TITLE
Scheduler

### PR DESCRIPTION
**Pull request by jimbofreedman**
Some spiders are killed for using too much memory, so we assign them two units for double the resources.

Some spiders take a long time to run, so we run them first - avoids us having 1699 complete and then waiting a day for `visa` to finish.

### CODE REVIEW

**Comment by ryantam626** [comment]
:eyes:

File: locations/commands/schedule_all.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    def run(self, args, opts):
+        project = args[0] if len(args) > 0 else "default"
+
+        spiders = self.crawler_process.spider_loader.list_schedule()
+        for spider, conf in spiders.items():
+            priority = conf["priority"]
+            units = conf["units"]
+            subprocess.run(
+                ["pipenv", "run", "shub", "schedule", f"-p {priority}", f"-u {units}", f"{project}/{spider}"]
```
Comment (ryantam626) [inline locations/commands/schedule_all.py:21]: Not sure if ` f"-p {priority}"` actually works, can you confirm it does?
I think it has to be 
`["pipenv", "run", "shub", "schedule", "-p", f"{priority}", .....]`

File: .github/workflows/spiders-scheduler.yaml [hunk 2]
```diff
@@ -30,9 +30,5 @@ jobs:
           pipenv install --dev
       - name: Schedule spiders
         shell: bash
-        # Currently reverse sort the list so that western_union, usps and visa
-        # run earlier and don't hold up the whole process at the end
-        # (long-term decision pending on excluding them)
-        # Priority 0 while we only have one container group - dev spiders will run first
         run: |
-          pipenv run scrapy list | xargs -n1 | sort -r | xargs -P 10 -I § pipenv run shub schedule -p 0 $SHUB_PROJECTID/§
+          pipenv run scrapy schedule_all ${{ secrets.scrapycloud_project_id }}
```
Comment (ryantam626) [inline .github/workflows/spiders-scheduler.yaml:34]: i am guessing `${{ secrets.scrapycloud_project_id }}` is the same as `$SHUB_PROJECTID` here?

File: locations/commands/schedule_all.py (hunk 1 above)
Comment (jimbofreedman) [reply locations/commands/schedule_all.py:21]:
```
(alltheplaces) ~/d/h/alltheplaces scheduler• 1.3s | 1 ❱ scrapy schedule_all                                                                                        14:22:28
/Users/jimbo/.local/share/virtualenvs/alltheplaces-TEO_jIsd/lib/python3.11/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
Spider 24_hour_fitness scheduled, job ID: 668973/1/16
Watch the log on the command line:
    shub log -f 1/16
or print items as they are being scraped:
    shub items -f 1/16
or watch it running in Scrapinghub's web interface:
    https://app.scrapinghub.com/p/668973/1/16
/Users/jimbo/.local/share/virtualenvs/alltheplaces-TEO_jIsd/lib/python3.11/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
Spider 2xu_au scheduled, job ID: 668973/2/12
Watch the log on the command line:
    shub log -f 2/12
or print items as they are being scraped:
    shub items -f 2/12
or watch it running in Scrapinghub's web interface:
    https://app.scrapinghub.com/p/668973/2/12
```

Comment (jimbofreedman) [reply locations/commands/schedule_all.py:21]: image
(temp changed to `zyte_units = 6` to check

File: .github/workflows/spiders-scheduler.yaml (hunk 2 above)
Comment (jimbofreedman) [reply .github/workflows/spiders-scheduler.yaml:34]: Good catch thanks!

File: locations/commands/schedule_all.py (hunk 1 above)
Comment (ryantam626) [reply locations/commands/schedule_all.py:21]: :+1:

**Comment by ryantam626** [comment]
LGTM

**Pull request merged**